src/lem/mod.rs

```diff
@@ -340,14 +350,26 @@ impl Func {
                         is_bound(src, map)?;
                         is_unique(tgt, map);
                     }
```

Comment:
Note for the `check()` function: the `#[inline(always)]` attributes are better replaces by `#[inline]`, the former is rarely appropriate.